MetaMask/metamask-extension
# feat(22139): add header to requests going to CoinGecko

## **Description**

Add the header to request sending to CoinGecko to help avoid api abuse.

## **Related issues**

Fixes: https://app.zenhub.com/workspaces/extension-delivery-board-6216892781ac020010e826d1/issues/gh/metamask/metamask-extension/22139

## **Manual testing steps**

1. Re-install the extension
2. Toggle auto detect token on
3. Import any token in mainnet
4. Check network request with "coin"
5. Validate that there's header as `{ X-Coingecko-Source: metamask.dec.jan.2024 }`

## **Screenshots/Recordings**



### **Before**
<img width="1285" alt="Screenshot 2023-12-04 at 21 45 38" src="https://github.com/MetaMask/metamask-extension/assets/12678455/b58ee59d-8e04-41d3-8b3b-1e31d670760b">


### **After**
<img width="1281" alt="Screenshot 2023-12-04 at 21 57 54" src="https://github.com/MetaMask/metamask-extension/assets/12678455/449c3b1c-29ee-4c0c-b411-341cc06b2551">


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
